app.go

```diff
@@ -138,7 +138,13 @@ type App struct {
 
 // New creates and initializes an App. All applications begin with the
 // Lifecycle type available in their dependency injection container.
+//
+// It then executes all functions supplied via the Invoke option. Supplying
+// arguments to these functions requires calling some of the constructors
+// supplied by the Provide option. If any invoked function fails, an error is
+// returned immediately.
 func New(opts ...Option) *App {
+	// TODO(glib): New should probably return (*App, error)
```

Comment:
My last resort :) https://github.com/golang/go/wiki/PanicAndRecover

By convention, no explicit panic() should be allowed to cross a package boundary. Indicating error conditions to callers should be done by returning error value.